pom.xml

```diff
@@ -12,11 +12,12 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 <!--        TODO: Move to new version of Java-->
-        <maven.compiler.source>11</maven.compiler.source>
-        <maven.compiler.target>11</maven.compiler.target>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
         <log4j.version>2.20.0</log4j.version>
         <grpc.version>1.63.0</grpc.version>
         <protobuf.version>3.25.3</protobuf.version>
```

Comment:
nit, don't really care at all but maybe use the BOM here? https://mvnrepository.com/artifact/io.github.hakky54/sslcontext-kickstart-bom